firewood/src/merkle.rs

```diff
@@ -1760,13 +1775,31 @@ mod tests {
     #[test_case(branch(b"value".to_vec(), vec![1, 2, 3].into()) ; "branch with chd")]
     #[test_case(branch(b"value".to_vec(), None); "branch without chd")]
     #[test_case(branch(Vec::new(), None); "branch without value and chd")]
-    fn node_encode_decode_(node: Node) {
+    fn node_encode_decode(node: Node) {
         let merkle = create_test_merkle();
         let node_ref = merkle.put_node(node.clone()).unwrap();
+        let expected_hash = node.get_root_hash(merkle.store.as_ref());
+
         let encoded = merkle.encode(&node_ref).unwrap();
         let new_node = Node::from(merkle.decode(encoded.as_ref()).unwrap());
         let new_node_hash = new_node.get_root_hash(merkle.store.as_ref());
+
+        assert_eq!(new_node_hash, expected_hash);
+    }
+
+    #[test_case(leaf(Vec::new(), Vec::new()) ; "empty leaf encoding")]
+    #[test_case(leaf(vec![1, 2, 3], vec![4, 5]) ; "leaf encoding")]
+    #[test_case(branch(b"value".to_vec(), vec![1, 2, 3].into()) ; "branch with chd")]
+    #[test_case(branch(Vec::new(), vec![1, 2, 3].into()); "branch without value")]
+    fn node_encode_decode_plain(node: Node) {
+        let merkle = create_test_merkle();
+        let node_ref = merkle.put_node(node.clone()).unwrap();
         let expected_hash = node.get_root_hash(merkle.store.as_ref());
+
+        let encoded = merkle.encode(&node_ref).unwrap();
+        let new_node = Node::from(merkle.decode(encoded.as_ref()).unwrap());
+        let new_node_hash = new_node.get_root_hash(merkle.store.as_ref());
+        
         assert_eq!(new_node_hash, expected_hash);
     }
```

Comment:
Thought about the same thing, was thinking need to make the test case generic to be able to do so, but I guess an option is also fine.

Reply:
Hmm, tried to do it but encountered difficulties, it seems this requires giving different types for `merkle`  based on condition. Do you have some suggestions?

Reply:
I think I see the issue here. I think you should delete the `Node::encode` method since we now have `Merle::Encode` (same goes for decode). 

Also, I don't think that checking the root hash is necessary. If `node == new_node`, then `node.get_root_hash(store) == new_node.get_root_hash(store)` when using the same `store`. You've already verified that the inputs to `get_root_hash` are the same, and I think it's overkill to test for determinism. Even if we do want to test for determinism, that should be done in a separate test.

Reply:
:+1, however I cannot remove `Node::encode` as the new encode/decode method doesn't cover extension node. https://github.com/ava-labs/firewood/blob/main/firewood/src/merkle.rs#L98

Reply:
Can we create an issue for this, please?

Reply:
https://github.com/ava-labs/firewood/issues/432